File: DownloadFiles/TGNet/Chapter06/Chapter6DemoCode/InheritanceDemoVB/InheritanceDemoVB/Form1.vb
Public Class Form1

  Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
    Dim x As New Class2
    x.Name = "Joe"
    MessageBox.Show(x.ShowName())
  End Sub
End Class

Public Class Class1
  Private strName As String

  Public Property Name() As String
    Get
      Return strName
    End Get
    Set(ByVal value As String)
      strName = value
    End Set
  End Property

  Public Function ShowName() As String
    Return "This is the name: " + Name
  End Function
End Class

Public Class Class2
  Inherits Class1

  Private strPhoneNumber As String

  Public Property PhoneNumber() As String
    Get
      Return strPhoneNumber
    End Get
    Set(ByVal value As String)
      strPhoneNumber = value
    End Set
  End Property

  'This is an overload because it has different parameters as the parent
  Public Function ShowName(ByVal message As String) As String
    Return "message: " + Name
  End Function

  'This is an Override because it has the same parameters as the parent
  Public Function ShowName() As String
    Return Name + ": " + PhoneNumber
  End Function


End Class

File: DownloadFiles/TGNet/Chapter06/Chapter6DemoCode/InheritanceDemoCS/InheritanceDemoCS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace InheritanceDemoCS
{
  public partial class Form1 : Form
  {
    public Form1()
    {
      InitializeComponent();
    }

    private void button1_Click(object sender, EventArgs e)
    {
      Class2 x = new Class2();
      x.Name = "Joe";
      MessageBox.Show(x.ShowName()); 
    }
  }

  public class Class1
  {
    private string strName;

    public string Name
    {
      get
      {
        return strName;
      }
      set
      {
        strName = value;
      }
    }

    public string ShowName()
    {
      return "This is the name: " + Name;
    }
  }
  public class Class2 : Class1
  {
    private string strPhoneNumber;

    public string PhoneNumber
    {
      get
      {
        return strPhoneNumber;
      }
      set
      {
        strPhoneNumber = value;
      }
    }

    //This is an overload because it has different parameters as the parent
    public string ShowName(string message)
    {
      return "message: " + Name;
    }
    //This is an Override because it has the same parameters as the parent
    public string ShowName()
    {
      return Name + ": " + PhoneNumber;
    }

  }


}